tests/test_all_rules.py

```diff
@@ -1417,3 +1423,67 @@ def test_new_terms_fields_unique(self):
             if rule.contents.data.type == "new_terms":
                 assert len(set(rule.contents.data.new_terms.value)) == len(rule.contents.data.new_terms.value), \
                     f"new terms fields values are not unique - {rule.contents.data.new_terms.value}"
+
+
+class TestESQLRules(BaseRuleTest):
+    """Test ESQL Rules."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up test environment."""
+        cls.dr_es_user = os.environ.get("DR_ES_USER")
+        cls.dr_cloud_id = os.environ.get("DR_CLOUD_ID")
+        cls.dr_elasticsearch_url = os.environ.get("DR_ELASTICSEARCH_URL")
+        cls.dr_validate_esql = os.environ.get("DR_VALIDATE_ESQL")
+
+        if cls.dr_validate_esql is None:
+            raise unittest.SkipTest("Test only run when DR_VALIDATE_ESQL environment variable set.")
+        assert cls.dr_es_user is not None, "DR_ES_USER environment variable is not set."
+        assert cls.dr_cloud_id is not None or cls.dr_elasticsearch_url is not None, \
+            "Either DR_CLOUD_ID or DR_ELASTICSEARCH_URL must be set."
+
+        super().setUpClass()
+
+    def run_esql_test(self, esql_query, expectation, message):
+        """Test that the query validation is working correctly."""
+        rc = RuleCollection()
+        file_path = Path(get_path("tests", "data", "command_control_dummy_production_rule.toml"))
+        original_production_rule = load_rule_contents(file_path)
+
+        # Test that a ValidationError is raised if the query doesn't match the schema
+        production_rule = deepcopy(original_production_rule)[0]
+        production_rule["rule"]["query"] = esql_query
+
+        expectation.match_expr = message
+        with expectation:
+            rc.load_dict(production_rule)
+
+    def test_esql_queries(self):
+        """Test ESQL queries."""
+        test_cases = [
+            # invalid queries
+            ('from .ds-logs-endpoint.events.process-default-* | wheres process.name like "Microsoft*"',
+             pytest.raises(marshmallow.exceptions.ValidationError), r"ESQL query failed"),
+            ('from .ds-logs-endpoint.events.process-default-* | where process.names like "Microsoft*"',
+             pytest.raises(marshmallow.exceptions.ValidationError), r"ESQL query failed"),
+
+            # valid queries
+            ('from .ds-logs-endpoint.events.process-default-* | where process.name like "Microsoft*"',
+             does_not_raise(), None),
+        ]
+        for esql_query, expectation, message in test_cases:
+            self.run_esql_test(esql_query, expectation, message)
+
+    def test_esql_rules(self):
+        """Test ESQL rules."""
+        for rule in self.production_rules:
+            if rule.contents.data.type == "esql":
+
+                # Stub test to validate esql rules
+                assert rule.contents.data.language == "esql", \
```

Comment:
this is already validated within the dataclass since we define language and type specific to the rule type